pandas/core/indexers.py

```diff
@@ -65,29 +69,24 @@ def is_empty_indexer(indexer, arr_value: np.ndarray) -> bool:
 def check_setitem_lengths(indexer, value, values) -> None:
     """
     Validate that value and indexer are the same length.
-
-    An special-case is allowed for when the indexer is a boolean array
-    and the number of true values equals the length of ``value``. In
-    this case, no exception is raised.
+    A special-case is allowed for when the indexer is a
+    boolean array, and the number of true values, equals to
+    the length of ``value``.
+    In this case, no exception is raised.
 
     Parameters
     ----------
     indexer : sequence
-        The key for the setitem
+        Key for the setitem.
     value : array-like
-        The value for the setitem
+        Value for the setitem.
     values : array-like
-        The values being set into
-
-    Returns
-    -------
-    None
+        Values being set into.
 
     Raises
     ------
     ValueError
-        When the indexer is an ndarray or list and the lengths don't
-        match.
+        When the indexer is a ndarray or list and the lengths don't match.
```

Comment:
```suggestion
        When the indexer is an ndarray or list and the lengths don't match.
```

"an" is actually correct here I think

Reply:
@jorisvandenbossche English is not my native language. All I know that's you put ```a``` before words that starts with ```A,E,I,O,U``` (probably missed a letter or two).
If it's such a big of a deal, I don't mind reverting, besides you shouldn't take my work for it, my grammar isn't great.

Reply:
I am also no native english speaker, but I know that the a vs an depends on the pronounciation, not spelling (see eg https://writingexplained.org/a-vs-an-difference). 
And in this case, I actually looked at numpy's docs, and they clearly use "an ndarray" much more than "a ndarray".

Reply:
Thank you so much for the patience!

Do you mind if I write a test case for that?